src/components/album/AlbumChat.js

```diff
@@ -49,17 +81,87 @@ const AlbumChat = ({ apiUrl }) => {
     };
   }, [albumId, socket]);
 
+  useEffect(() => {
+    scrollToBottom();
+  }, [messages]);
+
   return (
-    <div>
-      <h2>Album Chat</h2>
-      <div>
-        <input
-          type="text"
-          placeholder="Enter message"
-          value={message}
-          onChange={(e) => setMessage(e.target.value)}
-        />
-        <button onClick={handleSendMessage}>Send</button>
+    <div
+      style={{
+        height: '100%',
+        width: '100%',
+        display: 'flex',
+        flexDirection: 'column',
+      }}
+    >
+      <Typography variant="h6" component="div" style={{ marginLeft: '20px' }}>
+        Album Chat
+      </Typography>
+      <div
+        style={{
+          flex: 1,
+          padding: '16px',
+          width: '95%',
+          maxHeight: '65vh',
+          overflowY: 'auto',
+        }}
+        ref={messagesContainerRef}
+      >
+        {messages.map((message) => (
+          <div key={message._id}>
+            <div
+              style={{
+                display: 'flex',
+                flexDirection: message.isOwnMessage ? 'row-reverse' : 'row',
+                alignItems: 'center',
+                margin: '20px',
+              }}
+            >
+              <div
+                style={{
+                  backgroundColor: message.isOwnMessage
+                    ? '#3f51b5'
+                    : 'lightblue',
+                  color: message.isOwnMessage ? 'white' : 'black',
+                  borderRadius: '8px',
+                  padding: '8px 16px',
+                }}
+              >
+                <strong>{message.user.username}</strong>: {message.message}
```

Comment:
Your log line on line 57 is showing me that the messages that are being returned by the backend do not have a `message.user.username` field. I've left review comments in the backend to address that

![](https://github.com/Code-the-Dream-School/dd-prac-team4-front/assets/9094098/d75ce97d-14cc-48da-95c1-1ac54f6294e4)